test/cqlpy/test_custom_index.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+
+###############################################################################
+# Tests for custom indexes
+###############################################################################
+
+import pytest
+from .util import new_test_table
+from cassandra.protocol import InvalidRequest
+
+
+def test_create_vector_search_index(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE CUSTOM INDEX ON {table}(v) USING 'opensearch_vector_index'")
+
+def test_create_vector_search_index_with_usearch_engine(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE CUSTOM INDEX ON {table}(v) USING 'usearch_vector_index'")
+
+def test_create_vector_search_index_without_custom_keyword(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f"CREATE INDEX ON {table}(v) USING 'opensearch_vector_index'")
+
+def test_create_custom_index_with_invalid_class(cql, test_keyspace):
+    schema = 'p int primary key, v vector<float, 3>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        invalid_custom_class = "invalid.custom.class"
+        with pytest.raises(InvalidRequest):
+            cql.execute(f"CREATE CUSTOM INDEX invalid_idx ON {table}(v) USING '{invalid_custom_class}'")
```

Comment:
Nitpick: you don't need to specify an explict index name ("invalid_idx"). I think it just distraction to the test. I thought for a second that "invalid_idx" was the invalid thing, but it isn't, it's just a random name.